File: Wild.java
// これはワイルドカード型を使ったJavaクラス
import java.util.*;

public class Wild {
    public Collection<?> contents() {
        Collection<String> stuff = new Vector<String>();
        stuff.add("a");
        stuff.add("b");
        stuff.add("see");
        return stuff;
    }
}

File: Wild.scala
val contents = (new Wild).contents
contents.size()

import scala.collection.mutable
import java.util.Collection

abstract class SetAndType {
    type Elem
    val set: mutable.Set[Elem]
}

def javaSet2ScalaSet[T](jset: Collection[T]): SetAndType = {
    val sset = mutable.Set.empty[T]  // Tという名前で呼べるようになった

    val iter = jset.iterator
    while (iter.hasNext)
        sset += iter.next()
    
    new SetAndType {
        type Elem = T
        val set = sset
    }
}


